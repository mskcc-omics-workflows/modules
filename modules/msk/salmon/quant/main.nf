process SALMON_QUANT {
    tag "$meta.id"
    label "process_medium"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/salmon:0.14.0' :
        'docker.io/mskcc/salmon:0.14.0' }"


    input:
    tuple val(meta), path(reads)
    path  index

    output:
    tuple val(meta), path("${meta.id}.salmon.quant")    , emit: results
    tuple val(meta), path("${meta.id}.quant.sf")        , emit: quant
    path  "versions.yml"                                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args     = task.ext.args   ?: ''
    def prefix   = task.ext.prefix ?: "${meta.id}"
    def threads  = task.cpus * 2

    """
    salmon quant \\
        --index ${index} \\
        --seqBias \\
        --gcBias \\
        --libType A \\
        --unmatedReads ${reads} \\
        --validateMappings \\
        -o ${prefix}.salmon.quant \\
        ${args} \\
        --threads ${threads}

    cp ${prefix}.salmon.quant/quant.sf ${prefix}.quant.sf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        salmon: \$(echo \$(salmon --version) | sed -e "s/salmon //g")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir ${prefix}.salmon.quant
    touch ${prefix}.quant.sf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        salmon: \$(echo \$(salmon --version) | sed -e "s/salmon //g")
    END_VERSIONS
    """
}
