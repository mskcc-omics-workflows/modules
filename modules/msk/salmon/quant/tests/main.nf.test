nextflow_process {

    name "Test Process SALMON_QUANT"
    script "../main.nf"
    process "SALMON_QUANT"
    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "salmon"
    tag "salmon/quant"
    tag "salmon/index"

    setup {
        run("SALMON_INDEX") {
            script "../../../salmon/index/main.nf"
            process {
                """
                input = [file(params.test_data_mskcc['rediscoverte']['rollup_part_19_fa_gz'], checkIfExists: true)]
                """
            }
        }
    }

    test("rna - fastq") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end: true ], // meta map
                    [ file(params.test_data_mskcc['rediscoverte']['input_R1_fq_gz'], checkIfExists: true) ]
                ])
                input[1] = SALMON_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.quant[0][1]).name,
                    process.out.results[0].size()
                ).match() }
            )
        }

    }

    test("rna - fastq - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end: true ], // meta map
                    [ file('input_R1_fq_gz') ]
                ])
                input[1] = SALMON_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.quant[0][1]).name,
                    process.out.results[0].size()
                ).match() }
            )
        }

    }


}