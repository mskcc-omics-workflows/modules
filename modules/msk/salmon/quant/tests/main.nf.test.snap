{
    "rna - fastq - stub": {
        "content": [
            [
                "versions.yml:md5,6d0c92b7638e05c2385e21f67be1a5f4"
            ],
            "test.quant.sf",
            2
        ],
        "timestamp": "2024-06-28T13:18:39.004848"
    },
    "rna - fastq": {
        "content": [
            [
                "versions.yml:md5,6d0c92b7638e05c2385e21f67be1a5f4"
            ],
            "test.quant.sf",
            2
        ],
        "timestamp": "2024-06-28T13:18:28.714343"
    }
}